lib/spack/spack/cmd/log.py

```diff
@@ -0,0 +1,98 @@
+# Copyright 2013-2024 Lawrence Livermore National Security, LLC and other
+# Spack Project Developers. See the top-level COPYRIGHT file for details.
+#
+# SPDX-License-Identifier: (Apache-2.0 OR MIT)
+
+import os
+import tempfile
+import time
+
+import llnl.util.filesystem as fs
+import llnl.util.tty as tty
+
+import spack.cmd
+import spack.environment as ev
+import spack.util.compression as compression
+from spack.cmd.common import arguments
+
+description = "print out logs for packages"
+section = "basic"
+level = "long"
+
+
+def setup_parser(subparser):
+    subparser.add_argument(
+        "--continuous", action="store_true", help="for stage logs, poll for new content"
+    )
+
+    arguments.add_common_arguments(subparser, ["spec"])
+
+
+def dump_build_log(path, continuous):
+    with open(path, "r") as fstream:
+        line = fstream.readline()
+        while True:
+            if line:
+                print(line.strip())
+            elif continuous:
+                time.sleep(0.2)
```

Comment:
This appears to work for me on Mac OS, but I can't find any documentation that guarantees that it would (e.g. on Linux, I can't find conditions where an EOF indicator is unset other than when the user explicitly calls a function).

Reply:
I would honestly just execve tail -f

Reply:
> which may hit all the bugs already addressed over the decades in coreutils.

Do you have references to these? I'm asking because I'm interested in reading some of this. I couldn't find anything definitive about what is/is-not supported, or about what should/should-not be supported.

Overall this is generally punted on account of your observation in https://github.com/spack/spack/pull/42202#discussion_r1466230761. I'd like to add (in a follow-up PR) a mechanism for auto-detecting the phase logs as they are created and streaming them.

Reply:
Why would you choose the obviously more difficult way of doing things? Just make spack also output the combined logs next to the phase log, and follow that single file.

Reply:
I'm fine doing it that way instead if it's easier. I don't know if messing with the `log_output` interface is easier, perhaps in this case it wouldn't be so bad, e.g. modifying installer.py/log_output like:

```
    log_contextmanager = log_output(
                        log_file,
                        self.echo,
                        True,
                        env=self.unmodified_env,
                        filter_fn=self.filter_fn,
                        duplicate_to=other_log_file,      # <-------- we are already intercepting everything so we could likely write it twice without much worry
                    )
```

doing it this way makes me wish we weren't compressing the logs in the first place though, since then we could just supply the filenames and let the user `tail -f` it themselves.